mkdocs/docs/experiments/categories/load/k6-loadgen.md

```diff
@@ -0,0 +1,184 @@
+        name: k6-loadgen-sa
+        roleRef:
+        apiGroup: rbac.authorization.k8s.io
+        kind: Role
+        name: k6-loadgen-sa
+        subjects:
+        - kind: ServiceAccount
+          name: k6-loadgen-sa
+          namespace: default
+
+        ```
+        Use this sample RBAC manifest to create a chaosServiceAccount in the desired (app) namespace. This example consists of the minimum necessary role permissions to execute the experiment.
+
+## Experiment tunables
+
+??? info "check the experiment tunables" 
+    <h2>Optional Fields</h2>
+
+    <table>
+      <tr>
+        <th> Variables </th>
+        <th> Description  </th>
+        <th> Notes </th>
+      </tr>
+      <tr>
+        <td> TOTAL_CHAOS_DURATION  </td>
+        <td> The time duration for chaos injection (seconds) </td>
+        <td> Defaults to 20s </td>
+      </tr>
+      <tr>
+        <td> CHAOS_INTERVAL  </td>
+        <td> Time interval b/w two successive k6-loadgen (in sec) </td>
+        <td> If the CHAOS_INTERVAL is not provided it will take the default value of 10s </td>
+      </tr>
+      <tr>
+        <td> RAMP_TIME </td>
+        <td> Period to wait before injection of chaos in sec </td>
+        <td> </td>
+      </tr>
+      <tr>
+        <td> LIB_IMAGE  </td>
+        <td> LIB Image used to excute k6 engine </td>
+        <td> Defaults to <code>ghcr.io/grafana/k6-operator:latest-runner</code></td>
+      </tr>
+      <tr>
+        <td> LIBImagePullPolicy  </td>
+        <td> LIB Image pull policy </td>
+        <td> Defaults to <code>Always</code> </td> 
+      </tr>
+      <tr>
+        <td> SCRIPT_SECRET_NAME </td>
+        <td> Provide the k8s secret name of the JS script to run k6. </td>
+        <td> Default value: k6-script </td>
+      </tr>
+      <tr>
+        <td> SCRIPT_SECRET_KEY </td>
+        <td> Provide the key of the k8s secret named SCRIPT_SECRET_NAME </td>
+        <td> Default value: script.js </td>
+      </tr>
+    </table>
+
+## Experiment Examples
+
+### Common and Pod specific tunables
+
+Refer the [common attributes](../common/common-tunables-for-all-experiments.md) and [Pod specific tunable](common-tunables-for-pod-experiments.md) to tune the common tunables for all experiments and pod specific tunables.  
+
+### Custom k6 configuration
+You can add k6 options(ex hosts, thresholds) in the script `options` object. More details can be found [here](https://grafana.com/docs/k6/latest/using-k6/k6-options/)
+
+### Custom Secret Name and Secret Key
+
+You can provide the secret name and secret key of the JS script to be used for k6-loadgen. The secret should be created in the same namespace where the `chaos infrastructure` is created. For example, if the chaos infrastructure is created in the `litmus` namespace, then the secret should also be created in the `litmus` namespace. And If we want to use `custom-k6-script` secret and `custom-script.js` as the secret key, then the experiment tunable will look like this:
```

Comment:
@namkyu1999 , do you think it would help to add a short .js example snippet to provide more context to users? Or maybe some reference links to examples within the K6 website/docs.

Reply:
I added a sample JS script and also reference link:)